File: domain/src/main/java/com/ahn/domain/repository/FavoriteCurrencyRepository.kt
package com.ahn.domain.repository

import kotlinx.coroutines.flow.Flow

interface FavoriteCurrencyRepository {
    fun getFavoriteCurrencyCodes(): Flow<List<String>>
    suspend fun addFavorite(currencyCode: String)
    suspend fun removeFavorite(currencyCode: String)
    suspend fun isFavorite(currencyCode: String): Boolean
}

File: ios/Packages/Domain/Sources/Domain/Favorite/Repository/FavoriteCurrencyRepository.swift
public protocol FavoriteCurrencyRepository: Sendable {
    func favoriteCurrencyCodes() -> AsyncStream<[String]>

    func addFavorite(_ currencyCode: String) async throws

    func removeFavorite(_ currencyCode: String) async throws

    func isFavorite(_ currencyCode: String) async throws -> Bool
}
